src/Main.hs

```diff
@@ -119,7 +118,7 @@ closestIntersection :: Ray -> [Object] -> Maybe (Double, Object)
 closestIntersection ray objects
     | null intersections = Nothing
     | otherwise = Just $ minimumBy minimumDefinedByFirst intersections
-    where intersections = catMaybes $ map (intersects ray) objects
+    where intersections = mapMaybe (intersects ray) objects
```

Comment:
yep, exactly. it's a bit of a weird map function because the output list can be smaller than the input list, but it fits what we need :)